crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs

```diff
@@ -0,0 +1,476 @@
+use crate::{
+    context::LintContext,
+    fixer::Fix,
+    rule::Rule,
+    utils::{
+        collect_possible_jest_call_node, is_equality_matcher, parse_expect_jest_fn_call,
+        KnownMemberExpressionProperty, PossibleJestNode,
+    },
+};
+use oxc_ast::{
+    ast::{Argument, BinaryExpression, Expression},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use oxc_syntax::operator::BinaryOperator;
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "eslint-plugin-jest(prefer-comparison-matcher): Suggest using the built-in comparison matchers"
+)]
+#[diagnostic(severity(warning), help("Prefer using `{0:?}` instead"))]
+struct UseToBeComparison(&'static str, #[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferComparisonMatcher;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// This rule checks for comparisons in tests that could be replaced with one of the
+    /// following built-in comparison matchers:
+    /// - `toBeGreaterThan`
+    /// - `toBeGreaterThanOrEqual`
+    /// - `toBeLessThan`
+    /// - `toBeLessThanOrEqual`
+    ///
+    /// ### Examples
+    ///
+    /// ```js
+    /// // invalid
+    /// expect(x > 5).toBe(true);
+    /// expect(x < 7).not.toEqual(true);
+    /// expect(x <= y).toStrictEqual(true);
+    /// ```
+    ///
+    /// ```js /// // valid
```

Comment:
A small typo. Maybe the "// valid" is in next line

Reply:
Done. Nice Catch.